src/Core/Services/VaultTimeoutService.cs

```diff
@@ -67,14 +67,34 @@ public async Task<bool> IsLockedAsync(string userId = null)
             }
             return !hasKey;
         }
-        
+
+        public async Task<bool> ShouldLockAsync(string userId = null)
+        {
+            if (await ShouldTimeoutAsync(userId))
+            {
+                var action = await _stateService.GetVaultTimeoutActionAsync(userId);
+                return action == VaultTimeoutAction.Lock;
+            }
+            return false;
```

Comment:
This can be written as:
```c#
return await ShouldTimeoutAsync(userId) 
       && 
       await _stateService.GetVaultTimeoutActionAsync(userId) == VaultTimeoutAction.Lock;
```

Reply:
Fixed in #1830